source/opt/insert_extract_elim.cpp

```diff
@@ -74,6 +79,33 @@ bool InsertExtractElimPass::EliminateInsertExtract(ir::Function* func) {
             cid = cinst->GetSingleWordInOperand(kInsertCompositeIdInIdx);
             cinst = def_use_mgr_->GetDef(cid);
           }
+          // If search ended with CompositeConstruct or ConstantComposite
+          // and the extract has one index, return the appropriate component.
+          // If a vector CompositeConstruct we make sure all preceding
+          // components are of component type (not vector composition).
+          // TODO(greg-lunarg): Handle multiple-indices, ConstantNull, special
+          // vector composition, and additional CompositeInsert.
+          if ((cinst->opcode() == SpvOpCompositeConstruct ||
+               cinst->opcode() == SpvOpConstantComposite) &&
+              (*ii).NumInOperands() == 2) {
+            uint32_t compIdx = (*ii).GetSingleWordInOperand(1);
+            if (IsVectorType(cinst->type_id())) {
+              if (compIdx < cinst->NumInOperands()) {
+                uint32_t i = 0;
+                for (; i <= compIdx; i++) {
```

Comment:
I *think* this verification is unnecessary for valid SPIR-V. 
But ok to keep it.

I'd like to see the scenario that forces this code to exist.